src/test/test-mountpoint-util.c

```diff
@@ -64,6 +64,11 @@ static void test_mnt_id(void) {
                 path = NULL;
         }
 
+#if HAS_FEATURE_MEMORY_SANITIZER
+#  warning "msan doesn't grok sscanf(%ms)"
```

Comment:
If it's reproducible with the latest version of clang, could you open an issue on their bug tracker at https://github.com/google/sanitizers/issues?

As I said in https://github.com/systemd/systemd/pull/11730#issuecomment-466356765, I don't think it's good idea to continue working on that PR in its current form, but if you decided that it should go anyway, I think it would be better to reduce the scope of workarounds as much as possible. In the case of `%ms` it's probably possible to unpoison the argument after `sscanf` before it's used anywhere.

Reply:
There was an issue open for it certainly. I don't remember the number. But I don't have the latest versions of the llvm stack installed, so it's quite possible that this is already fixed upstream.

I'll reply to the more general question in #11730.

I updated the PR to use __msan_unpoison. The test still passes locally.

Reply:
Regarding `sscanf`, the false positive was reported in https://github.com/google/sanitizers/issues/767 and it's still reproducible with clang 9.0.0 (trunk 353250).